lsp-go.el

```diff
@@ -55,12 +55,14 @@ completing function calls."
   :risky t
   :package-version '(lsp-mode "6.2"))
 
-(defcustom lsp-gopls-env (make-hash-table)
+(defcustom lsp-gopls-env '()
   "`gopls' has the unusual ability to set environment variables,
   intended to affect the behavior of commands invoked by `gopls'
   on the user's behalf. This variable takes a hash table of env
   var names to desired values."
-  :type '(restricted-sexp :match-alternatives (hash-table-p))
+  :type '(repeat (cons (string :tag "env var name") (string :tag "value")))
+  :set 'lsp--defcustom-set-hashtable-from-set
+  :get 'lsp--defcustom-get-set-from-hashtable
```

Comment:
`:get` and `:set` should not be needed.

Reply:
Also :get/:set are introduced in emacs 26 and we target emacs 25 as well.

Reply:
Indeed! I can simply use the `alist` as-is. I saw in the trace output:
```
Params: {
  "jsonrpc": "2.0",
  "id": 1,
  "result": [
    {
      "env": {
        "asdf": "asdf"
      },
      "buildFlags": [],
      "hoverKind": "SynopsisDocumentation",
      "usePlaceholders": true
    },
    null
  ]
}
```
When setting env var and value to `asdf` and `asdf` respectively.

Reply:
Yep. Not related to the PR but what is the server request? Because one of the sections is null.

Reply:
Ah actually I still need to transform `nil` (false) to `:json-false` otherwise I get:
```
[Trace - 10:28:48 PM] Received notification 'window/showMessage'.
Params: {
  "message": "Invalid type %!d(string=test) for map key \"codelens\" in option %!q(MISSING)",
  "type": 1
}
```

Reply:
You should be setting this in the actual value. I mean, you should not set it to nil but to :json-false. We cannot safely assume that nil is false.